tests/foreman/ui/test_contentview.py

```diff
@@ -4123,3 +4123,44 @@ def test_positive_depsolve_with_module_errata(session, module_org):
             assert items['Name'] in rpm_pack
         assert result['module_streams']['table'][0]['Name'] == mod_stream
         assert result['errata']['table'][0]['Errata ID'] == settings.repos.yum_10.errata[0]
+
+
+@pytest.mark.tier2
+def test_positive_filter_by_pkg_group_name(session, module_org):
+    """Publish a filtered version of a Content View filtered using package group name.
+
+    :id: c7021f46-0168-44f7-a863-4aa34533efdb
+
+    :expectedresults: Published Content View's contents is filtered
+                based on package group.
+
+    :CaseImportance: Medium
+    """
+
+    filter_name = gen_string('alpha')
+    repo_name = gen_string('alpha')
+    package_group = 'birds'
+    expected_packages = (('cockateel'), ('duck'), ('penguin'), ('stork'))
+    create_sync_custom_repo(module_org.id, repo_name=repo_name)
+    repo = entities.Repository(name=repo_name).search(query={'organization_id': module_org.id})[0]
+    cv = entities.ContentView(organization=module_org, repository=[repo]).create()
+    with session:
+        session.contentviewfilter.create(
+            cv.name,
+            {
+                'name': filter_name,
+                'content_type': FILTER_CONTENT_TYPE['package group'],
+                'inclusion_type': FILTER_TYPE['include'],
+            },
+        )
+        session.contentviewfilter.add_package_group(cv.name, filter_name, package_group)
+        cvf = session.contentviewfilter.read(cv.name, filter_name)
+        assert cvf['content_tabs']['assigned'][0]['Name'] == package_group
+        # Publish CV and check filter worked
+        session.contentview.publish(cv.name)
+        result = session.contentview.read_version(cv.name, VERSION)
+        # Assert we do not have more than the expected number of packages
+        assert len(result['rpm_packages']['table']) == 4
+        # Assert the package name are the expected ones
+        for items in result['rpm_packages']['table']:
```

Comment:
Consider making a more direct comparison here that provides a bit more context on failure.

Removing the for loop, this also makes checking against a hardcoded length number redundant, since the list equality will ensure that there are no extraneous or missing packages.
```
assert expected_packages == [pkg['Name'] for pkg in result['rpm_packages']['table']]
```

Reply:
HI, `expected_packages` is a tuple (same as some other tests), so I was thinking `expected_packages = tuple([pkg['Name'] for pkg in result['rpm_packages']['table']])`, but then it seems neater just to change `expected_packages` to a list. Doing that now.